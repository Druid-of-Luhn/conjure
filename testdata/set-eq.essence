language Essence 2.0

find x,y : set (size 4) of int(0..9-1)

such that
    x = y,
    3 elem x,
    $ 4 elem y,
    $ 6 elem y,
    $ 5 elem x,
