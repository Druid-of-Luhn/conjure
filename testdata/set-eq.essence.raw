Spec
  { language = "Essence"
  , version = [ 2
              , 0
              ]
  , topLevelBindings = [ ( Find
                         , "x"
                         , DomainSet
                             { size = Just ( ValueInteger 4 )
                             , minSize = Nothing
                             , maxSize = Nothing
                             , attrDontCare = False
                             , element = DomainIntegerFromTo (
                                           Just ( ValueInteger 0 ) ) (
                                           Just ( ValueInteger 9 ) )
                             , representation = Nothing
                             }
                         )
                       , ( Find
                         , "y"
                         , DomainSet
                             { size = Just ( ValueInteger 4 )
                             , minSize = Nothing
                             , maxSize = Nothing
                             , attrDontCare = False
                             , element = DomainIntegerFromTo (
                                           Just ( ValueInteger 0 ) ) (
                                           Just ( ValueInteger 9 ) )
                             , representation = Nothing
                             }
                         )
                       ]
  , topLevelWheres = []
  , objective = Nothing
  , constraints = [ GenericNode
                      Eq
                      [ Identifier "x"
                      , Identifier "y"
                      ]
                  ]
  , metadata = []
  }