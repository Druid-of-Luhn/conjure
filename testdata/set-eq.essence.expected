language Essence 2.0

find x : set (size 4) of int(0..9 - 1)
find y : set (size 4) of int(0..9 - 1)

such that
    x = y,
    3 elem x
